src/request/sign-transaction/SignTransaction.js

```diff
@@ -11,29 +72,44 @@ class SignTransaction { // eslint-disable-line no-unused-vars
      * @returns {Promise<SignedTransactionResult>}
      * @protected
      */
-    async _signTx(txRequest, passphraseOrPin) {
-        if (txRequest.type !== TransactionType.BASIC) {
-            throw new Error('Not yet implemented');
+    async _doSignTransaction(txRequest, passphraseOrPin) {
+        if (txRequest.type === TransactionType.BASIC) {
+            // eslint-disable-next-line object-curly-newline
+            const { value, fee, recipient, signer, validityStartHeight } = txRequest;
+
+            const key = await KeyStore.instance.get(signer, passphraseOrPin);
+            const tx = key.createTransaction(recipient, value, fee, validityStartHeight);
+
+            const signatureProof = Nimiq.SignatureProof.unserialize(new Nimiq.SerialBuffer(tx.proof));
+
+            return {
+                type: TransactionType.BASIC,
+                sender: tx.sender.toUserFriendlyAddress(),
+                signerPubKey: signatureProof.publicKey.serialize(),
+                recipient: tx.recipient.toUserFriendlyAddress(),
+                value: Nimiq.Policy.satoshisToCoins(tx.value),
+                fee: Nimiq.Policy.satoshisToCoins(tx.fee),
+                validityStartHeight: tx.validityStartHeight,
+                signature: signatureProof.signature.serialize(),
```

Comment:
why unserialize and then serialize the proof again?

Reply:
Because the `proof` is stored in a `Nimiq.Transaction` object as an `Uint8Array`, and thus needs to be deserialized to be a `Nimiq.SignaturProof` (line 84). To transfer it via a postMessage, it needs to be a primitive type again, and postMessage can work with Uint8Arrays, so to transfer over RPC we serialize the signature again into an Uint8Array.